File: 2021/go/days/d01.go
package days

import (
	"fmt"
	"strconv"
	"strings"

	h "github.com/moymat/aoc2021/helpers"
)

func computeNbOfIncrease(file string) int {
	lines := strings.Split(h.GetInput("d01", file), "\n")
	total := 0

	for idx := range lines {
		if idx == 0 {
			continue
		}

		if idx == len(lines)-2 {
			break
		}

		previous, err := strconv.Atoi(strings.TrimSpace(lines[idx-1]))
		if err != nil {
			panic("not a number")
		}

		next, err := strconv.Atoi(strings.TrimSpace(lines[idx+2]))
		if err != nil {
			panic(("not a number"))
		}

		if next > previous {
			total += 1
		}
	}

	return total
}

func RunD01(file string) {
	fmt.Println(computeNbOfIncrease(file))
}

File: 2021/ts/d01.ts
import { getInput } from "./helpers";

function computeNbOfIncrease() {
  return getInput()
    .split("\n")
    .map((_, i, arr) =>
      i === 0 || i === arr.length - 2 ? 0 : +arr[i + 2] - +arr[i - 1]
    )
    .filter((nb) => nb > 0).length;
}

console.log(computeNbOfIncrease());
